src/goto-checker/single_path_symex_only_checker.cpp

```diff
@@ -37,29 +37,24 @@ operator()(propertiest &properties)
 {
   resultt result(resultt::progresst::DONE);
 
-  // First run goto-symex from entry point to initialize worklist
+  auto get_goto_function =
+    [this](const irep_idt &id) -> const goto_functionst::goto_functiont & {
+    return goto_model.get_goto_function(id);
+  };
```

Comment:
Could this also be refactored? Code copies are undesirable...

Reply:
Yes... unfortunately, C++ lambdas are so clumsy that a one-line lambda feels like code duplication...

Reply:
Shouldn't we really have an interface that doesn't require passing this parameter? I believe that in fact we already do, in `goto_symext::symex_with_state`, which has a variant with and without that function. So maybe we could also do that for those other functions?

Reply:
> variant without that function
That one should actually be removed.

Reply:
Ok, then maybe that could happen here as well? I dislike code copies, but I even more dislike almost-code-copies which we currently have with `get_function_from_goto_functions`.

Reply:
Check the changes, please.